### PR TITLE
Reformat CLI interface

### PR DESCRIPTION
**Pull request by alxjrvs**
In the interest of keeping the CLI a little flexible, I've updated the interface to have two new root commands: `init` and `add`. 

Stealing from the Readme: 

---
## Commands

There are two commands provided by `Gnar-cli`: `init` and `add`.

### gnar init

You can use `init` to greenfield new projects with specific frameworks.

```bash
$ gnar init rails new-app
```

This is equivalent to running `rails new new-app <gnarly config options here>`.

Right now, `init` supports `rails`.

### gnar add

You can use `add` to configure already-existing projects with specific configurations for specific dependencies.

```bash
$ gnar add eslint
```

This will add `eslint` to the `package.json` of the local project, as well as modifying it with Gnarly Opinions.

Right now, `add` supports `eslint` and `prettier`.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -7,20 +7,39 @@ import { Eslint, Prettier, Rails } from './commands'
...
-  .argument('<mainCommand>', 'first argument passed to rails')
-  .argument('[subcommand]', 'Second command passed to rails')
-  .description('install ESLint with Gnarly configuration - now with Prettier!')
-  .action(Rails.run)
+    switch (frameworkName.toLowerCase()) {
+      case 'rails':
+        Rails.init(subCommand, extraArgs)
+        break
+      default:
+        process.stdout.write(`Unrecognized framework name passed to gnar add: ${frameworkName}`)
```
Comment (efertsch) [inline src/index.ts:40]: Should we list the permissible framework names?

File: src/index.ts [hunk 2]
```diff
@@ -7,20 +7,39 @@ import { Eslint, Prettier, Rails } from './commands'
...
+  .action(packageName => {
+    switch (packageName.toLowerCase()) {
+      case 'prettier':
+        Prettier.add()
+        break
+      case 'eslint':
+        Eslint.add()
+        break
+      default:
+        process.stdout.write(`Unrecognized package name passed to gnar add: ${packageName}`)
```
Comment (efertsch) [inline src/index.ts:22]: Should we list the permissible package names?

File: src/commands/rails/index.ts [hunk 3]
```diff
@@ -5,6 +5,23 @@ import { Gnarrc } from '../../utils/gnarrc'
 export class Rails {
   static command = 'rails'
 
+  static async init(name: string, extraArgs: string) {
+    process.stdout.write(`Setting up Rails New with Gnarly opinions...\n`)
```
Comment (efertsch) [inline src/commands/rails/index.ts:9]: We print some crazy gnar beast ASCII art to the terminal 😏

**Review by efertsch — APPROVED** [review]
Great work, I like this interface a lot 😄 I left a couple of comments about listing acceptable frameworks/package names but this is GTG otherwise.

File: src/index.ts (hunk 1 above)
Comment (alxjrvs) [reply src/index.ts]: Listed!

File: src/index.ts (hunk 2 above)
Comment (alxjrvs) [reply src/index.ts]: Listed!

File: src/commands/rails/index.ts (hunk 3 above)
Comment (alxjrvs) [reply src/commands/rails/index.ts]: I had only low-quality gnar beast images, but I think this can be added to the GNIP board as an easy first project!

**Pull request merged**